### ISSUE TITLE
Patch 2.0.5 create huge issue with asynchronous Validator (standard or custom)

### ISSUE THREAD

**Issue by Dinuz** [joecorcoran/judge#44]
@joecorcoran 

Hi Joe,
This is kind of super weird.
I follow your suggestions to add a custom validation (it is an asyncronous validator - need to perform a call to the server in order to check the presence of a specific parameter in a model).
What is weird is that without the custom validator, all the other standard validator present on the model work as a charm, instead when i include my custom validator, it just run it and not more the standard validator.

Maybe I am doing something very wrong, and I need some clarifications or help about it.

The follow are the step that I did:

(1) Created my custom validator in rails (it works fine I tested it on the server side).
(2) Added uses_messages :validator_error_name to use the i18n with judge
(3) Created my custom validator client side (I put it in a js file called judge-extensions, and I included it in my app.js after judge) :+1: 

    judge.eachValidators.user_existence = function() {
      // create a 'pending' validation
      var validation = new judge.Validation();
      $.ajax(judge.urlFor(this, 'user_existence')).done(function(messages) {
      // You can close a Validation with either an array
      // or a string that represents a JSON array
      validation.close(messages);
      });
      return validation;
    };

(4) I added the expose Model, :parameter in the config for the judge

(5) I added my custom validator to my model

    validates :email, presence: true, 
    format: {with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i},
    user_existence: true

Now When I run my code, I see that the custom validator is called, and it works fine (gives error when it need to give error, and give valid when need to be valid).
This means that the code is right, except that it become the only validation that runs.
I mean with my custom validation, format and presence are completely ignored (are not performed at all, like are not present in the model).
If I comment out in the model the custom validator, then the other two are automatically working again.

What I am doing wrong? Because I am doing something wrong for sure.

Thanks so much

Best
Dinuz

**Comment by joecorcoran** [comment]
Let's try and get to the bottom of this. I *think* I understand but I'm finding it a bit hard to follow. I thought [the spec](https://github.com/joecorcoran/judge/blob/master/spec/javascripts/judge-spec.js#L94) already covered this, but I might well be wrong.

Can you put together a brand new Rails app with Judge 2.0.5 and the most basic example of this behaviour possible? That would be really helpful. Please try to use plain callback functions in JS wherever you can, so we can eliminate function binding as the source of the problem.

**Comment by Dinuz** [comment]
Thanks joe, I will do it tomorrow!
Let's understand this thing well!

**Comment by Dinuz** [comment]
@joecorcoran did you have the opportunity to look at the app the I put in the public repository?
If yes, could you reproduce the erroneous behaviour that I experimented?

Did you see that the data-object, has multiple repetition of the same error message?
And that when the custom validator is activate, the other no ajax validation are completely ignored?
The js that I used is pretty basic, and is just integrated with bootstrap style.

But also if I use the most basic callback (in that repository)  like:

      $(function() {
           //$('form').JudgeBootstrapValidator(); 
           $('#invite_request_email').on('blur', function() {                
                judge.validate(document.getElementById('invite_request_email'), {
                     valid: function(element) {
                                   element.style.border = '1px solid green';
                      },
                      invalid: function(element, messages) {
                                  element.style.border = '1px solid red';
                                 alert(messages.join(','));
                       }
                   });
               }); 
            });

The effect is the same:

When there are ajax validation, all the other validation are always skipped.

This should already prove that the binding is not the source of the problem.

On top of this there is also the issue of the validation run client side, also if a field is not present in that specific form, if the validation is activate in the model.

I gave you the example of the sign-in (and I agreed with you was not the best example), but was important anyway to highlight the weird behaviour. In that specific case, the validations were always failing, because judge was running a password_confirmation validation. and it couldn't find any password confirmation, because was a sign in (note that the same model has a password confirmation, used for example in a sign up form, or in a change password form).

**Comment by dannysperry** [comment]
@Dinuz @joecorcoran, I can confirm that this bug is definitely happening. I set up a basic app to test #38 against. and I ended up setting presence and uniqueness on the attribute. This and #42 seem to be the same issue. I'm just about finished up on #38 and then i was going to start digging into this. As @Dinuz mentioned before #23 might also be tied to this. but i haven't given that issue enough attention to say for sure
